application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -514,15 +535,34 @@ define('diagram-store', ['jquery', 'xwiki-meta', 'xwiki-utils', 'diagram-utils',
         // See comment from care 'li'.
         alt.removeAttribute('text-anchor');
         break;
+      case 'u':
+        alt.setAttribute('text-decoration', 'underline');
+        break;
       case 'a':
         var link = document.createElementNS('http://www.w3.org/2000/svg', 'a');
         link.setAttribute('x', w);
         link.setAttribute('y', h);
         link.setAttributeNS('http://www.w3.org/1999/xlink', 'xlink:href', element.href);
-        // Add link decorations for the text.
-        alt.setAttribute('fill', '#0000EE');
-        alt.setAttribute('text-decoration', 'underline');
+        // Add link decorations for the text. Some styles added to a 'g' are not distributed to it's children by FOP,
+        // so we do it manually.
+        if (alt.tagName.toLowerCase() == 'g') {
+          var childNodes = $(alt).children();
+          childNodes.each(function(index) {
+            var child = childNodes[index];
```

Comment:
I think you can use ``this`` to access the child node inside the iterator function.